lib/pluginUtils.js

```diff
@@ -127,15 +128,22 @@ function checkVersion (fn) {
 
 function registerPluginName (fn) {
   const meta = getMeta(fn)
-  if (!meta) return
 
-  const name = meta.name
-  if (!name) return
+  const name = meta?.name ?? 'anonymous'
```

Comment:
not 100% sure here - I remember some logic somewhere where we read the `function.toString()` payload when the name is empty and setting it as `anonymous` may bypass that chack, need to check.

Instead of pushing the string everytime, I would just return the `name` (if any) and then pass it to `checkPluginHealthiness(fn, name || 'anonymous')`